lib/experimental/blocks.php

```diff
@@ -77,3 +77,46 @@ function wp_enqueue_block_view_script( $block_name, $args ) {
 		add_filter( 'render_block', $callback, 10, 2 );
 	}
 }
+
+/*
+ * WP_Block_Styles_Registry was marked as `final` in core so it cannot be
+ * updated via Gutenberg to allow registration of a style across multiple
+ * block types as well as with an optional style object. This function will
+ * support the desired functionality until the styles registry can be updated
+ * in core.
+ */
+if ( ! function_exists( 'gutenberg_register_block_style' ) ) {
+	/**
+	 * Registers a new block style for one or more block types.
+	 *
+	 * @param string|array $block_name       Block type name including namespace or array of namespaced block type names.
+	 * @param array        $style_properties Array containing the properties of the style name, label,
+	 *                                       style_handle (name of the stylesheet to be enqueued),
+	 *                                       inline_style (string containing the CSS to be added),
+	 *                                       style_data (theme.json-like object to generate CSS from).
+	 *
+	 * @return bool True if all block styles were registered with success and false otherwise.
+	 */
+	function gutenberg_register_block_style( $block_name, $style_properties ) {
+		if ( ! is_string( $block_name ) && ! is_array( $block_name ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				__( 'Block name must be a string or array.', 'gutenberg' ),
+				'5.3.0'
+			);
+
+			return false;
+		}
+
+		$block_names = is_string( $block_name ) ? array( $block_name ) : $block_name;
+		$result      = true;
+
+		foreach ( $block_names as $name ) {
+			if ( ! WP_Block_Styles_Registry::get_instance()->register( $name, $style_properties ) ) {
+				$result = false;
```

Comment:
The `$result` isn't being set to true at all in the loop. 

Initially, it is set to `true`, then any failure to register a block style flips it to `false`. 

So I believe it does satisfy `"True if all block styles were registered with success and false otherwise."`

Am I understanding the question correctly?

Reply:
Nope, I'm reading it wrong. Thanks for the explainer.